tests/spec/conftest.py

```diff
@@ -41,6 +43,11 @@ def run_test_execute(table_data, series, expected_results, population=None):
         assert len(results) == len(results_dict), "Duplicate patient IDs found"
         assert results_dict == expected_results
 
+        # assert types are as expected
+        variables = compile(dataset)
+        variable_type = get_series_type(variables["v"])
+        assert results[0][1] is None or isinstance(results[0][1], variable_type)
```

Comment:
This only checks the first result, which is not unreasonable as they should all have the same type unless something _really_ horrible has happened. But, if the first value happens to be `None` then it's not going to do any type-checking at all.

You could make it check the first non-null value, but it's probably simpler to check the type of every value than it is to do that.

Reply:
Addressed in latest commit (I hope!)